index.js

```diff
@@ -22,7 +22,7 @@ module.exports = exports = function fetch(input, init = {}) {
 
   return promise
 
-  async function process(input) {
+  async function process(input, crossOriginRedirect = false) {
```

Comment:
Pass the origin of the redirect, i.e. `request._url` further down, here as `origin` so we avoid parsing `res.headers.location` prematurely, twice, and outside the appropriate `try` block.